learn/work/fairness-detectors/index.qmd

```diff
@@ -0,0 +1,290 @@
+
+```{r roc-aucs}
+detectors %>%
+  group_by(detector) %>%
+  roc_auc(truth = kind, .pred_AI) %>%
+  arrange(desc(.estimate)) %>%
+  head(3)
+```
+
+::: callout-tip
+To learn more about how the yardstick package handles groups in data, see the ["Grouping behavior in yardstick" vignette](https://yardstick.tidymodels.org/dev/articles/grouping.html).
+:::
+
+<!-- TODO: remove the /dev from the above link once this yardstick vignette makes it to CRAN -->
+
+From this perspective, the models with the highest `roc_auc()` estimates are the most fair.
+
+The fairness assessment for this stakeholder doesn't need any of the functionality newly introduced in yardstick 1.3.0. An analysis that reconciles the role of the `native` variable in these models' predictions will, though.
+
+## Fair prediction on human-written essays
+
+Now, consider that our only priority was to correctly classify human-written text as human-written and incorrectly classify human-written text as generated by AI at the same rate for both native English writers and non-native English writers, ignoring predictions from essays generated by AI.
+
+-   Another **student** whose work is evaluated by a detector model may take on such a perspective. This student could be a native English writer who does not want other students to be subjected to undue harm or a non-native English writer concerned with their own writing being incorrectly classified as GPT-generated.
+
+-   Another **instructor** of a course may feel it is unfair to disproportionately classify writing from students that are non-native English writers as GPT-generated, regardless of how effectively the model detects GPT-generated essays.
+
+The `equal_opportunity()` metric enables us to quantify the extent of this interpretation of unfairness. Equal opportunity is satisfied when a model's predictions have the same true positive and false negative rates across protected groups; a model predicts more fairly if it's equally likely to predict a positive outcome for each group.
+
+In this example, a GPT detector satisfies equal opportunity when the detector correctly classifies human-written text as human-written and incorrectly classifies human-written text as generated by AI at the same rate for both native English writers and non-native English writers.
+
+::: callout-note
+This definition does not consider the predictions based on essays generated by AI.
+:::
+
+To calculate equal opportunity with yardstick, we use the `equal_opportunity()` function:
+
+```{r ep-opp-native}
+equal_opportunity_by_native <- equal_opportunity(by = native)
+```
+
+<!-- TODO: print out the following function once the metric print method is merged into yardstick -->
+
+The function `equal_opportunity_by_native()` is a yardstick metric function like any other, except it knows to temporarily group by and summarize across a data-column called `native`. Applying it:
+
+```{r eq-opp}
+detectors %>%
+  filter(kind == "Human") %>%
```

Comment:
We shouldn't have to include this step, I think, as this metric doesn't use the predictions on essays generated by AI. For some reason, this errors out at the moment.

Reply:
Closing in light of looking into this at https://github.com/tidymodels/yardstick/issues/464.